### PR TITLE
Test for URL and IP canonicalization

### PR DESCRIPTION
**Pull request by AlbaMontis**

### CODE REVIEW

File: tests/testwebfiles.js [hunk 1]
```diff
@@ -17,6 +17,20 @@
 
 
 /**
+ * Waits when data will be loaded.
+ */
+function waitLoading() {
```
Comment (vpodk) [inline tests/testwebfiles.js]: Where this function is used?

File: tests/testothermetrics.js [hunk 2]
```diff
@@ -13,7 +13,7 @@
     return !!~(['ok', 'malware', 'unknown'].indexOf(content));
   },
   /** @return {boolean} */ 'mozdata-links': function(content) {
-    return PATTERN.test(content) || 'N/A' == content;
+    return PATTERN.test(content) || 'N/A' == content || '-' == content;
```
Comment (vpodk) [inline tests/testothermetrics.js]: Linked sites can be numeric or N/A:

``` javascript
formatter_.formatNumber(~~data['links'] || 0) : 'N/A';
```

**Comment by vpodk** [comment]
Name of this pull request is "Test for URL and IP canonicalization", but it contain also functionality not related to canonicalization.

**Comment by Alexsv7** [comment]
LGTM

File: tests/testothermetrics.js (hunk 2 above)
Comment (RomanFedytskyi) [inline tests/testothermetrics.js]: fixed

File: tests/testcanonicalization.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+ *  "display: none"
+ * -tag with "failClass" should contain inner text "No" and have style
+ *  "display: block" or "display: inline-block"
+ */
+
+
+/**
+ * @return {boolean} Returns "true" if test failed.
+ */
+function testCanonicalization() {
```
Comment (vpodk) [inline tests/testcanonicalization.js:19]:
``` javascript
/**
 * @return {boolean} Returns "true" if test failed.
 */
function testCanonicalization() {
  /** @type {boolean} */ var result = true;
  /** @type {Array.<string>} */ var widgets = ['canonical-url', 'canonical-ip'];
  /** @type {number} */ var length = widgets.length;
  /** @type {Element} */ var element;
  /** @type {Element} */ var passElement;
  /** @type {Element} */ var failElement;
  /** @type {Element} */ var passDisplay;
  /** @type {Element} */ var failDisplay;

  for (; length;) {
    element = document.getElementById(widgets[--length]);
    if (element) {
      passElement = element.getElementsByClassName('pass')[0];
      failElement = element.getElementsByClassName('fail')[0];
      if (passElement && failElement) {
        passDisplay = getComputedStyle(passElement).getPropertyValue('display');
        failDisplay = getComputedStyle(failElement).getPropertyValue('display');
        // Validate display property.
      }
    }
  }
  return result;
}
```

File: tests/testcanonicalization.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
+/**
+ * @fileoverview Defines tests for URL and IP canonalization.
+ * Successful criteria:
```
Comment (vpodk) [inline tests/testcanonicalization.js]: Format suggestion:

dash, space, Text, semicolon
dash, space, Text, semicolon
dash, space, Text, dot

Comment (RomanFedytskyi) [inline tests/testcanonicalization.js]: fixed

File: tests/testcanonicalization.js (hunk 3 above)
Comment (RomanFedytskyi) [inline tests/testcanonicalization.js:19]: fixed

File: tests/testcanonicalization.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+ * - canonicalization widget should exist;
```
Comment (vpodk) [inline tests/testcanonicalization.js]: https://en.wikipedia.org/wiki/Sentence_(linguistics)

Comment (RomanFedytskyi) [inline tests/testcanonicalization.js]: fixed

**Comment by mshchygol** [comment]
LGTM 👍

**Comment by vasylsemiliak** [comment]
LGTM

**Comment by Alexsv7** [comment]
LGTM

**Pull request closed without merging**